hudi-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataPayload.java

```diff
@@ -368,22 +367,11 @@ private HoodieMetadataColumnStats combineColumnStatsMetadatat(HoodieMetadataPayl
     }
 
     if (filesystemMetadata != null) {
-      filesystemMetadata.forEach((filename, fileInfo) -> {
-        // If the filename wasnt present then we carry it forward
-        if (!combinedFileInfo.containsKey(filename)) {
-          combinedFileInfo.put(filename, fileInfo);
-        } else {
-          if (fileInfo.getIsDeleted()) {
-            // file deletion
-            combinedFileInfo.remove(filename);
-          } else {
-            // file appends.
-            combinedFileInfo.merge(filename, fileInfo, (oldFileInfo, newFileInfo) -> {
-              return new HoodieMetadataFileInfo(oldFileInfo.getSize() + newFileInfo.getSize(), false);
-            });
-          }
-        }
-      });
+      validatePayload(type, filesystemMetadata);
+      // Combine previous record w/ the new one
+      // NOTE: New records _always_ take precedence over the old one, as such no special case
+      //       handling is actually necessary
+      combinedFileInfo.putAll(filesystemMetadata);
```

Comment:
This does not take care of deletes. 

If the filesystemMetadata has a delete, we want to remove it rather than adding a delete record into combinedFileInfo.

Reply:
@prashantwason can you please elaborate on why we want to have this behavior?

Reply:
Scrap that comment, figured out what you're referring to